habana/ubi8-python-3.8/Dockerfile

```diff
@@ -0,0 +1,171 @@
+        habanalabs-firmware-tools-"$VERSION"-"$REVISION".el8 \
+        habanalabs-graph-"$VERSION"-"$REVISION".el8 && \
+    rm -f /etc/yum.repos.d/habanalabs.repo && rm -rf /tmp/* && \
+    dnf clean all && rm -rf /var/cache/yum
+
+# There is no need to store pip installation files inside docker image
+ENV PIP_NO_CACHE_DIR=on
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1
+
+# Install python packages 
+# RUN python3.8 -m pip install hpu_media_loader=="${VERSION}"."${REVISION}"
+# Install Python packages and Jupyterlab extensions from Pipfile.lock
+COPY Pipfile.lock ./
+
+RUN echo "Installing softwares and packages" && micropipenv install && rm -f ./Pipfile.lock
+
+# SSH configuration necessary to support mpi-operator v2
+RUN mkdir -p /var/run/sshd && \
+    sed -i 's/[ #]\(.*StrictHostKeyChecking \).*/ \1no/g' /etc/ssh/ssh_config && \
+    sed -i 's/#\(ForwardAgent \).*/\1yes/g' /etc/ssh/ssh_config && \
+    echo "    UserKnownHostsFile /dev/null" >> /etc/ssh/ssh_config && \
+    sed -i 's/#\(StrictModes \).*/\1no/g' /etc/ssh/sshd_config && \
+    ssh-keygen -A && \
+    mkdir -p /var/run/sshd && echo "/usr/sbin/sshd -p 3022" | tee -a ~/.bashrc
+
+RUN echo "export LANG=en_US.UTF-8" >> /root/.bashrc
+RUN export LANG=en_US.UTF-8
+ENV GC_KERNEL_PATH=/usr/lib/habanalabs/libtpc_kernels.so
+ENV HABANA_LOGS=/var/log/habana_logs/
+ENV HABANA_SCAL_BIN_PATH=/opt/habanalabs/engines_fw
+ENV HABANA_PLUGINS_LIB_PATH=/opt/habanalabs/habana_plugins
+
+## Install habana pytorch
+ENV LANG=en_US.UTF-8
+ENV PYTHONPATH=/root:/usr/lib/habanalabs/
+
+RUN dnf install -y \
+    curl \
+    cairo-devel \
+    numactl-devel \
+    iproute \
+    which \
+    zlib-devel \
+    lapack-devel \
+    openblas-devel \
+    numactl \
+    gperftools-devel && \
+    dnf clean all && rm -rf /var/cache/yum
+
+RUN wget --no-verbose https://"${ARTIFACTORY_URL}"/artifactory/gaudi-pt-modules/"${VERSION}"/"${REVISION}"\
+/pytorch/rhel86/pytorch_modules-v"${PT_VERSION}"_"${VERSION}"_"${REVISION}".tgz && \
+    mkdir /root/habanalabs /root/habanalabs/pytorch_temp && \
+    tar -xf pytorch_modules-v"${PT_VERSION}"_"${VERSION}"_"${REVISION}".tgz -C /root/habanalabs/pytorch_temp/. && \
+    pip3 install mpi4py==3.1.4 && \
+    pip3 install /root/habanalabs/pytorch_temp/*.whl && \
+    pip3 install $(grep "pytorch-lightning" /root/habanalabs/pytorch_temp/requirements-pytorch.txt) && \
+    pip3 install -U habana-lightning-plugins=="${VERSION}"."${REVISION}.1" && \
+    /sbin/ldconfig && \
+    echo "source /etc/profile.d/habanalabs.sh" >> ~/.bashrc && \
+    pip3 uninstall -y pillow && \
+    pip3 uninstall -y pillow-simd && \
+    pip3 install pillow-simd==7.0.0.post3 && \
+    rm -rf /root/habanalabs/pytorch_temp/ && \
+    rm -rf pytorch_modules-v"${PT_VERSION}"_"${VERSION}"_"${REVISION}".tgz
+ENV LD_PRELOAD=/lib64/libtcmalloc.so.4
+ENV TCMALLOC_LARGE_ALLOC_REPORT_THRESHOLD=7516192768
+
+RUN dnf clean all && rm -rf /var/cache/dnf && rm -rf /tmp/*
+
```

Comment:
Regarding the Tensorflow installer, it appears that its instructions are not included in this Dockerfile.

Reply:
I have tried to install the most possible packages from Pipfile. so we can manage them.